internal/osbuild2/stage.go

```diff
@@ -128,6 +128,8 @@ func (stage *Stage) UnmarshalJSON(data []byte) error {
 		options = new(TunedStageOptions)
 	case "org.osbuild.sfdisk":
 		options = new(SfdiskStageOptions)
+	case "org.osbuild.sgdisk":
+		options = new(SgdiskStageOptions)
```

Comment:
Why do we have this?  I'm trying to remember  the purpose of unmarshalling stage options.
It's a bigger conversation, for another time, but I think it was because osbuild would return the options of a stage with the results.  I think it doesn't do that anymore (not for v2 manifests) but also, I don't think we use them anywhere.

Maybe it's time we got rid of this?

Reply:
Yeah, I didn't check it in detail, but I also think that we do not use this anywhere. +1 for getting rid of the whole UnmarshalJSON method completely 😉

Reply:
Done in https://github.com/osbuild/osbuild-composer/pull/2741